tests/test_select.py

```diff
@@ -242,20 +250,23 @@ def test_select_from_unnest_parsed_as_keyword():
     )
 
 
+# deactivated for sqlfluff since it can not be parsed properly
 def test_select_from_unnest_with_ordinality():
```

Comment:
This will make presto/trino user unhappy. Let's revisit all the unsupported cases later together.

Reply:
Trino and Presto is still not supported by `sqlfluff` but opening a PR to support them should not be hard.

Reply:
I just checked sqlfluff [dialect docs](https://docs.sqlfluff.com/en/stable/dialects.html), and it supports Athena. Athena is built on open-source Trino and Presto. So we can point Presto/Trino user to dialect="athena" for most cases.

Athena supports UNNEST: https://docs.aws.amazon.com/athena/latest/ug/flattening-arrays.html
But I don't see WITH CARDINALITY as mentioned in Presto docs supported in Athena: https://prestodb.io/docs/current/sql/select.html

If WITH CARDINALITY is really Presto only, then we have to walk-around. Otherwise, we can improve sqlfluff athena dialects.

Do you happen to have a Athena environment to try this SQL?

Reply:
I have reviewed it on our Athena instance, and it works, so we can contribute to `sqlfluff` for fixing it and point Presto/Trino users to the `athena` dialect.

<img width="979" alt="image" src="https://user-images.githubusercontent.com/11074908/220603488-24c731c9-4158-4cef-b971-4f5769ba9f39.png">

Reply:
Sorry I missed this thread. It's great news. I'll document this together with other things we'd want to improve sqlfluff as action items post this PR merge.